src/main/java/org/apache/flink/benchmark/InputBenchmark.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.benchmark;
+
+import org.apache.flink.benchmark.functions.LongSource;
+import org.apache.flink.benchmark.functions.MultiplyByTwo;
+import org.apache.flink.streaming.api.datastream.DataStreamSource;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.streaming.api.functions.sink.DiscardingSink;
+
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.OperationsPerInvocation;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jmh.runner.options.VerboseMode;
+
+@OperationsPerInvocation(value = InputBenchmark.RECORDS_PER_INVOCATION)
+public class InputBenchmark {
+	public static final int RECORDS_PER_INVOCATION = 7_000_000;
+
+	public static void main(String[] args)
+		throws RunnerException {
+		Options options = new OptionsBuilder()
+			.verbosity(VerboseMode.NORMAL)
+			.include(".*" + InputBenchmark.class.getSimpleName() + ".*")
+			.build();
+
+		new Runner(options).run();
+	}
+
+	@Benchmark
+	public void benchmarkChained(FlinkEnvironmentContext context) throws Exception {
+
+		StreamExecutionEnvironment env = context.env;
+		DataStreamSource<Long> source = env.addSource(new LongSource(RECORDS_PER_INVOCATION));
+
+		env.setParallelism(1);
+		source
+			.map(new MultiplyByTwo())
+			.addSink(new DiscardingSink<>());
+
+		env.execute();
+	}
+
+	@Benchmark
+	public void benchmarkChannel(FlinkEnvironmentContext context) throws Exception {
+
+		StreamExecutionEnvironment env = context.env;
```

Comment:
enable checkpointing in both benchmarks? The tricky part will be to find a good checkpointing interval value, since those benchmarks are executed within 500ms-1000ms.  

If we drop the `benchmarkCount`, we would loose non checkpointed coverage and in that case the question is does it matter for us? If after enabling checkpointing interval the throughput doesn't change by much, it probably doesn't matter.

Reply:
Just to be clear, would you like to enable it in general, or have it somehow parameterized/separate benchmark case? In my head, that almosts tests two different aspects of the system.

Reply:
100ms checkpointing interval should be fine. We can adjust it in the future if we see the need.